vivarium/core/engine.py

```diff
@@ -415,12 +418,19 @@ def __init__(
             progress_bar: shows a progress bar
             emit_config: If True, this will emit the serialized
                 processes, topology, and initial state.
+            profile: Whether to profile the simulation with cProfile.
         """
+        if profile:
+            self.profiler = cProfile.Profile()
+            self.profiler.enable()
```

Comment:
You need `self.profiler = None` perhaps if `profile=False`

Reply:
looks like you took car of this.

Reply:
I fixed this when fixing some failing tests